src/synapse/azext_synapse/custom.py

```diff
@@ -168,12 +181,30 @@ def delete_spark_pool(cmd, client, resource_group_name, workspace_name, spark_po
     return sdk_no_wait(no_wait, client.delete, resource_group_name, workspace_name, spark_pool_name)
 
 
-def create_sql_pool(cmd, client, resource_group_name, workspace_name, sql_pool_name, location, sku_name,
-                    sku_tier=None, max_size_bytes=None, source_database_id=None, recoverable_database_id=None,
-                    create_mode=None, tags=None, no_wait=False):
+def create_sql_pool(cmd, client, resource_group_name, workspace_name, sql_pool_name, sku_name, source_database_id=None,
+                    recoverable_database_id=None, tags=None, no_wait=False):
+    # get the location of the workspace
+    workspace_client = cf_synapse_client_workspace_factory(cmd.cli_ctx)
+    workspace_object = workspace_client.get(resource_group_name, workspace_name)
+    location = workspace_object.location
+
+    max_size_bytes = None
+    sku_tier = None
     sku = Sku(tier=sku_tier, name=sku_name)
     sql_pool_info = SqlPool(sku=sku, location=location, max_size_bytes=max_size_bytes,
                             source_database_id=source_database_id, recoverable_database_id=recoverable_database_id,
-                            create_mode=create_mode, tags=tags)
+                            tags=tags)

```

Comment:
Why do you remove `create_mode`?

Reply:
> Why do you remove `create_mode`?

there is no `create_mode` in `az sql db create`

Reply:
But you can see it in the ARM template when creating SQL pool through Azure portal.

Reply:
> But you can see it in the ARM template when creating SQL pool through Azure portal.

Like powershell, making it internally.